checker/tests/tainting/Issue4170.java

```diff
@@ -0,0 +1,49 @@
+import java.util.ArrayList;
+import org.checkerframework.checker.tainting.qual.Tainted;
+import org.checkerframework.checker.tainting.qual.Untainted;
+
+// @below-java10-jdk-skip-test
+public class Issue4170 {
+  public void method1() {
+    var list = new ArrayList<@Untainted String>();
+    ArrayList<@Untainted String> list2 = list;
```

Comment:
There are not test case that reassign a variable that was declared with var.   For example:
```
import java.util.List;
import org.checkerframework.checker.tainting.qual.Tainted;
import org.checkerframework.checker.tainting.qual.Untainted;

public class Var {
  void method(List<@Tainted String> taintedlist, List<@Untainted String> untaintedList ) {
    var list1 = taintedlist;
    List<@Tainted String> l = list1;
    // :: error: (assignment)
    list1 = untaintedList; 
  }
}
```
I think it might be surprising to the user that the reassignment fails, so we may want to add something to the manual to explain this.

Reply:
@smillst do you think a new sub-section of [the type refinement section](https://checkerframework.org/manual/#type-refinement) would be an appropriate place to document this?

Reply:
I think so.  I'll assign this to @mernst though to get his opinion.